x-pack/plugins/security_solution/public/management/components/console/types.ts

```diff
@@ -54,43 +56,57 @@ export interface CommandDefinition {
 /**
  * A command to be executed (as entered by the user)
  */
-export interface Command {
+export interface Command<TDefinition extends CommandDefinition = CommandDefinition> {
   /** The raw input entered by the user */
   input: string;
   // FIXME:PT this should be a generic that allows for the arguments type to be used
   /** An object with the arguments entered by the user and their value */
   args: ParsedCommandInput;
-  /** The command defined associated with this user command */
-  commandDefinition: CommandDefinition;
+  /** The command definition associated with this user command */
+  commandDefinition: TDefinition;
 }
 
-/**
- * The component that will handle the Command execution and display the result.
- */
-export type CommandExecutionComponent = ComponentType<{
-  command: Command;
+export interface CommandExecutionComponentProps<
+  TStore extends object = Record<string, unknown>,
+  TMeta = any
```

Comment:
I tried and was not able to get it to work in a reliable way. since these are not really used by the Console (they are just "pass-through" to the component that handles the execution of the console command), we really don't care here what its set to. Thus why I turned off the `any` rule at the top of the file